Bottom: 170bd5a990ef76c111a06c974a5d93d8325ac0fd
Top:    ccbb844e878a9e0775d96c7da39b9ee405996af8
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-12-26 23:29:46 +0000

hid/gtk: Refactor storage semantics of view_x, view_y coordinates

They now directly represent actual cursor coordinates, and don't require
passing through SIDE_{X,Y} before use.

NB: FUBAR ON ITS OWN.. TOO MANY PLACES ASSUME THE OLD WAY!


---

diff --git a/src/hid/gtk/gui-output-events.c b/src/hid/gtk/gui-output-events.c
index e0120b6..e40f7d2 100644
--- a/src/hid/gtk/gui-output-events.c
+++ b/src/hid/gtk/gui-output-events.c
@@ -221,8 +221,8 @@ ghid_get_coords (const char *msg, int *x, int *y)
     ghid_get_user_xy (msg);
   if (ghid_port.has_entered)
     {
-      *x = SIDE_X (gport->view_x);
-      *y = SIDE_Y (gport->view_y);
+      *x = gport->view_x;
+      *y = gport->view_y;
     }
 }
 
@@ -242,11 +242,10 @@ ghid_note_event_location (GdkEventButton * ev)
       event_x = ev->x;
       event_y = ev->y;
     }
-  gport->view_x = event_x * gport->zoom + gport->view_x0;
-  gport->view_y = event_y * gport->zoom + gport->view_y0;
+  gport->view_x = VIEW_X (event_x);
+  gport->view_y = VIEW_Y (event_y);
 
-  moved = MoveCrosshairAbsolute (SIDE_X (gport->view_x), 
-				 SIDE_Y (gport->view_y));
+  moved = MoveCrosshairAbsolute (gport->view_x, gport->view_y);
   if (moved)
     {
       AdjustAttachedObjects ();
